FirebaseUI/Database/Implementation/FirebaseDataSource.m

```diff
@@ -21,7 +21,14 @@
 #import "FirebaseDataSource.h"
 
 @interface FirebaseDataSource ()
-@property(copy, nonatomic) void (^cancelBlock)(NSError *);
+
+@property(copy, nonatomic, nonnull) void (^cancelBlock)(NSError *);
+
+/**
+ * The FirebaseArray which backs the instance of the datasource.
+ */
+@property(strong, nonatomic, nonnull) FirebaseArray *array;
```

Comment:
Made this private because it has all sorts of behaviors. Previously it was possible to accidentally break a `FirebaseDataSource` by assigning the delegate of the array to something else.